workers/unity/Assets/Playground/Scripts/Worker/AndroidClientWorkerConnector.cs

```diff
@@ -33,18 +33,22 @@ protected override void HandleWorkerConnectionEstablished()
                 return;
             }
 
-            levelInstance = Instantiate(level, transform);
-            levelInstance.transform.SetParent(null);
+            levelInstance = Instantiate(level, transform.position, transform.rotation);
         }
 
-        protected override void HandleWorkerConnectionFailure()
+        protected override void HandleWorkerConnectionFailure(string errorMessage)
         {
-            ConnectionScreenController.OnConnectionFailed();
+            ConnectionScreenController.OnConnectionFailed(errorMessage);
         }
 
         protected override string GetHostIp()
         {
 #if UNITY_ANDROID
+            if (Application.isEditor)
```

Comment:
Is this if block necessary? It looks like it would unnecessarily prevent connecting to different machines from the scene in editor

Reply:
as discussed internally: we don't want android clients to run in the editor if they didn't select Android as active platform to make sure everything runs as expected

Reply:
but if they select android as active platform `Application.isEditor` would be true I think, by "this block" I mean overwriting GetHostIP for editor no matter what you input into the field

I had just tested connecting to my mac from my pc which was running the Android client scene (when I had Android as selected platform) for example, with this change that would not be possible

Reply:
ahhhh, I see. so we could check, if we are running in the editor and the field is empty?

Reply:
think I misunderstodd the question :)

Reply:
I definitely want that block, because otherwise we sometimes pass in null as an ip adress. While it does seem to work, I still find that pretty dodgy

Reply:
Agreed, checking if field is empty or null, we can return to default too (preferably for all platforms?)